multiwallet.go

```diff
@@ -570,13 +629,13 @@ func (mw *MultiWallet) DeleteBadWallet(walletID int) error {
 	}
 
 	os.RemoveAll(wallet.dataDir)
-	delete(mw.badWallets, walletID)
+	mw.badWalletsDelete(walletID)
 
 	return nil
 }
 
 func (mw *MultiWallet) WalletWithID(walletID int) *Wallet {
-	if wallet, ok := mw.wallets[walletID]; ok {
+	if wallet, ok := mw.walletsReadCopy()[walletID]; ok {
```

Comment:
No need to copy the entire map to access just one wallet, lock the mutex here, find and return the desired wallet, then unlock the mutex on return.